boa/src/syntax/parser/mod.rs

```diff
@@ -121,10 +121,19 @@ where
     type Output = StatementList;
 
     fn parse(self, cursor: &mut Cursor<R>) -> Result<Self::Output, ParseError> {
-        if cursor.peek(0)?.is_some() {
-            ScriptBody.parse(cursor)
-        } else {
-            Ok(StatementList::from(Vec::new()))
+        match cursor.peek(0)? {
+            Some(tok) => {
+                match tok.kind() {
+                    TokenKind::StringLiteral(string) | TokenKind::TemplateLiteral(string) => {
+                        if string == &"use strict".into() {
```

Comment:
Could this line be changed to `string.as_str() == "use strict"`? I think it looks cleaner.

Reply:
.as_str won't work because string is of type Box<str> but .to_string() will and I agree it would be cleaner

Reply:
But `.to_string()` will result in an allocation, right? I'm looking into how to avoid it.

Reply:
`&*string == "use strict"` should work fine.
Taken from here: https://stackoverflow.com/a/32085042

Reply:
Changed to .as_ref()

Reply:
> `&*string == "use strict"` should work fine.
> Taken from here: https://stackoverflow.com/a/32085042

Hmm, I wonder what the performance difference is to .as_ref() - &* seems like strange notation!

Reply:
It is strange notation, but I think the codegen should be equal.
My concern was avoiding alocation, and I don think `.as_ref()` is clearer.